cmdstanpy/install_cmdstan.py

```diff
@@ -306,7 +272,7 @@ def retrieve_version(version: str, progress: bool = True) -> None:
             if progress:
                 progress_hook: Optional[
                     Callable[[int, int, int], None]
-                ] = wrap_progress_hook()
+                ] = wrap_url_progress_hook()
```

Comment:
This currently does nothing for me in console or jupyter

Reply:
this works for me when I invoke the script itself from the terminal:
```
python3 cmdstanpy/install_cmdstan.py --overwrite --progress
```
but it doesn't work when it's invoked via utils function `install_cmdstan`.  not sure what the problem is.

Reply:
My guess is TQDM detects if it is being redirected/piped and disables itself. Probably finally biting us that install_cmdstan the function really just uses subprocess to call install_cmdstan the script. With the recent refactor it shouldn't be too hard to change this if we wanted to, we could just write a main function which takes in a dict and call it directly from the code. For the script, we could just call `main(argparse.parse(sys.argv))` or similar

Reply:
I think this is worthwhile to fix but I can do that in a later PR if that's easier

Reply:
Yeah it would actually be pretty simple, we do more or less the same thing to call the install_cxx_toolchain script from inside install_cmdstan. We do it by overwriting sys.argv though, which is a bit of a hack. It would be better to simply make `main()` take in the list and then we can call it in the `if __name__ == "__main__"` block with `sys.argv[1:]`

Reply:
removed option.